### PR TITLE
[Internal]: Fixes NullRef in TransportHandler when client is concurrently disposed.

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Pull Request Template

## Description

Fixes a race conditions where StoreProxy returned can be null when the client is getting disposed at the same time.

NOTE: Testing this is not easy due to this ebing a race condition. I evaluated using the ExceptionLess tests (which is using MockTransportHandler) - but it was not sutainable - because the race condition to hit the NullRef is tiny. Given that throwing ObjectDisposedException happens in other places regularly in this code path I think it is acceptable to not add a unit tests trying to provoke this edge condition.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs [hunk 1]
```diff
@@ -108,6 +108,19 @@ internal async Task<ResponseMessage> ProcessMessageAsync(
...
+            if (storeProxy == null)
+            {
+                // storeProxy being null should indicate that there was a race condition and the Client was
+                // disposed between getting teh DocumentClient and calling GetStoreProxy
+                // in this case repeating this call will result in throwing an ObjectDisposedException
+                storeProxy = this.client.DocumentClient.GetStoreProxy(serviceRequest);
+
+                if (storeProxy == null)
+                {
+                    throw new InvalidOperationException("StoreProxy cannot be null");
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs:120]: Thsi should never happen - I ledt this as a different Exception to avoid hiding an actual issue behind ObjectDisposedException. In the race of cleint being disposed - the line above would trigger the ObjectDisposedException - if the storeProxy is still null here - without throing ObjectDisposedException - then there is another case where stroe model is null.

**Review by kundadebdatta — APPROVED** [review]
LGTM! I agree coming up with a test is very tricky to replicate the race condition.

**Review by kirankumarkolli — APPROVED** [review]
